File: front_end/src/app/Services/acte-dentaire.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { ActeDentaire } from '../Models/Prestation';

@Injectable({
  providedIn: 'root'
})
export class ActeDentaireService {

  private apiUrl = 'http://localhost:8034/api/actes' ;


  constructor(private http: HttpClient) { }

  createActeDentaire(acteDentaire: ActeDentaire): Observable<ActeDentaire> {
    return this.http.post<ActeDentaire>(this.apiUrl, acteDentaire);
  }

  getActeDentaireById(id: number): Observable<ActeDentaire> {
    return this.http.get<ActeDentaire>(`${this.apiUrl}/${id}`);
  }

  getAllActeDentaires(): Observable<ActeDentaire[]> {
    return this.http.get<ActeDentaire[]>(this.apiUrl);
  }

  updateActeDentaire(id: number, acteDentaire: ActeDentaire): Observable<ActeDentaire> {
    return this.http.put<ActeDentaire>(`${this.apiUrl}/${id}`, acteDentaire);
  }

  deleteActeDentaire(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }

}

File: src/main/java/com/example/projectPfe/services/Interface/ActeDentaireService.java
package com.example.projectPfe.Services.Interface;

import com.example.projectPfe.models.ActeDentaire;

import java.util.List;
import java.util.Optional;

public interface ActeDentaireService {

    ActeDentaire createActeDentaire(ActeDentaire acteDentaire);
    ActeDentaire updateActeDentaire(int id, ActeDentaire acteDentaire);
    void deleteActeDentaire(int id);
    ActeDentaire getActeDentaireById(int id);
    List<ActeDentaire> getAllActeDentaires();

}
